yaptide/admin/simulators.py

```diff
@@ -70,35 +79,134 @@ def install_simulator(name: SimulatorType) -> bool:
+        if all([endpoint, access_key, secret_key, encryption_key]):
+            click.echo('Downloading from S3 bucket')
+            shieldhit_installed = download_shieldhit_from_s3()
+        if not shieldhit_installed:
+            click.echo('Downloading demo version from shieldhit.org')
+            download_shieldhit_demo_version()
     else:
         click.echo('Not implemented')
         return False
     return True
 
 
+def download_shieldhit_demo_version() -> bool:
+    """Download shieldhit demo version from shieldhit.org"""
+    demo_version_url = 'https://shieldhit.org/download/DEMO/shield_hit12a_x86_64_demo_gfortran_v1.0.1.tar.gz'
+    # check if working on Windows
+    if os.name == 'nt':
+        demo_version_url = 'https://shieldhit.org/download/DEMO/shield_hit12a_win64_demo_v1.0.1.zip'
+
+    # create temporary directory and download
+    # Create a temporary file to store the downloaded binary data
+    with tempfile.TemporaryDirectory() as tmpdir_name:
+        click.echo(f"Downloading from {demo_version_url} to {tmpdir_name}")
+        headers = {'User-Agent': 'Mozilla/5.0 (Windows NT x.y; rv:10.0) Gecko/20100101 Firefox/10.0'}
+        response = requests.get(demo_version_url, headers=headers)
+        temp_file_archive = Path(tmpdir_name) / Path(demo_version_url).name
+        with open(temp_file_archive, 'wb') as file_handle:
+            file_handle.write(response.content)
+        click.echo(f"Saved to {temp_file_archive} with size {temp_file_archive.stat().st_size} bytes")
+
+        # extract
+        click.echo(f"Extracting {temp_file_archive} to {installation_path}")
+        if temp_file_archive.suffix == '.gz':
+            extract_shieldhit_from_tar_gz(temp_file_archive, Path(tmpdir_name), 'shieldhit')
+        elif temp_file_archive.suffix == '.zip':
+            extract_shieldhit_from_zip(temp_file_archive, Path(tmpdir_name), 'shieldhit.exe')
+    return True
+
+
+def download_shieldhit_from_s3() -> bool:
+    """Download shieldhit from S3 bucket"""
+    s3_client = boto3.client(
+        "s3",
+        aws_access_key_id=access_key,
+        aws_secret_access_key=secret_key,
+        endpoint_url=endpoint
+    )
+    destination_file_path = installation_path / "shieldhit"
+    with tempfile.TemporaryFile() as temp_file:
+        # Download file from s3 bucket
+        try:
+            s3_client.download_fileobj("shieldhit", "shieldhit", temp_file)
+        except ClientError as e:
+            click.echo("S3 download failed with error: ", e.response["Error"]["Message"])
+            return False
+        temp_file.seek(0)
+        encrypted_data = temp_file.read()
+    # Decrypt downloaded file
+    click.echo("Decrypting downloaded file")
+    fernet = Fernet(encryption_key)
+    decrypted_data = fernet.decrypt(encrypted_data)
+    with open(destination_file_path, "wb") as f:
+        f.write(decrypted_data)
+    # Permission to execute
+    os.chmod(destination_file_path, 0o700)
```

Comment:
```suggestion
    # Decrypt downloaded file
    click.echo("Decrypting downloaded file")
    fernet = Fernet(encryption_key)
    decrypted_data = fernet.decrypt(encrypted_data)
    with open(destination_file_path, "wb") as f:
        f.write(decrypted_data)
    # Permission to execute
    os.chmod(destination_file_path, 0o700)
```

I would move that section entirely inside `with tempfile.TemporaryFile() as temp_file:` construct